addons/website_slides/models/slide_slide.py

```diff
@@ -959,12 +947,16 @@ def _action_set_quiz_done(self, completed=True):
                      slide.quiz_second_attempt_reward,
                      slide.quiz_third_attempt_reward,
                      slide.quiz_fourth_attempt_reward]
-            points += gains[user_membership_sudo.quiz_attempts_count - 1] if user_membership_sudo.quiz_attempts_count <= len(gains) else gains[-1]
-
-        if not completed:
-            points *= -1
+            points = gains[min(user_membership_sudo.quiz_attempts_count, len(gains)) - 1]
+            if points:
+                if completed:
+                    reason = _('Quiz Completed')
+                else:
+                    points *= -1
+                    reason = _('Quiz Set Uncompleted')
+                karma_added = self.env.user.sudo()._add_karma(points, slide, reason)
```

Comment:
Seems strange to set "karma_added" to True a lot of times, as there are no cases where it's set to False, maybe we should just return True at the end of this method and remove the dependency to karma?